client/src/components/Collections/common/services.ts

```diff
@@ -0,0 +1,8 @@
+import { type components, fetcher } from "@/api/schema";
+
+export type HistoryContentBulkOperationPayload = components["schemas"]["HistoryContentBulkOperationPayload"];
+
+const putHistories = fetcher.path("/api/histories/{history_id}/contents/bulk").method("put").create();
+export async function updateHistories(historyId: string, changes: HistoryContentBulkOperationPayload) {
```

Comment:
Can you rename this function and fetcher to something like `updateHistoryItemsInBulk` it will represent better what it is doing.
Also please move this fetcher to `client/src/api/histories.ts` so it can be easily discovered and reused in other places.